frappe/utils/nestedset.py

```diff
@@ -59,15 +59,16 @@ def update_add_node(doc, parent, parent_field):
 
 	# get the last sibling of the parent
 	if parent:
-		left, right = frappe.db.sql("select lft, rgt from `tab{0}` where name=%s"
-			.format(doctype), parent)[0]
-		validate_loop(doc.doctype, doc.name, left, right)
+		records = frappe.db.sql("select lft, rgt from `tab"+doctype+"` where name=%s", parent, as_dict=1)
+		if records:
+			left = records[0]['lft']
+			right = records[0]['rgt']
+			validate_loop(doc.doctype, doc.name, left, right)
+		else:
+			frappe.throw(_("Parent("+parent+") is invalid."))
 	else: # root
-		right = frappe.db.sql("""
-			SELECT COALESCE(MAX(rgt), 0) + 1 FROM `tab{0}`
-			WHERE COALESCE(`{1}`, '') = ''
-		""".format(doctype, parent_field))[0][0]
-	right = right or 1
+		right = frappe.db.sql("select ifnull(max(rgt),0)+1 from `tab%s` \
```

Comment:
Revert back to...
```suggestion
		right = frappe.db.sql("""
			SELECT COALESCE(MAX(rgt), 0) + 1 FROM `tab{0}`
			WHERE COALESCE(`{1}`, '') = ''
		""".format(doctype, parent_field))[0][0] #nosec
```
**Reason:**  `tab%s` will not work (it will be converted to eg. tab'Employee' ) ... since %s marker is used to add values with required quotes(for escaping) around them and not for identifiers...

use #nosec to avoid codacy warning here